apps/dot-voting/app/store/votes.js

```diff
@@ -226,8 +229,76 @@ const marshallVote = ({
     totalParticipation: totalParticipation,
     metadata,
     executionScript,
+    executionTargetData,
     executed,
     participationPct:
       totalVoters === 0 ? 0 : (totalParticipation / totalVoters) * 100,
   }
 }
+
+const getVoteExecutionTargets = (vote) => {
+  return vote.executionTargets || []
+}
+
+const decorateVote = async (vote) => {
+  return new Promise(resolve => {
+    const executionTargets = getVoteExecutionTargets(vote)
+    let targetApp
+    if (!executionTargets.length) {
+      // If there's no execution target, consider it targetting this Dot Voting app
+      app.currentApp()
+        .pipe(first())
+        .subscribe((currentApp) => {
+          resolve({
+            ...vote,
+            executionTargetData: {
+              ...currentApp,
+              identifier: undefined,
+            }
+          })
```

Comment:
I know we have a lot of the Promise-resolve pattern in our event store code, but I think it's important to utilize the rxjs libraries more effectively, like we've started to do in [Allocations](https://github.com/AutarkLabs/open-enterprise/blob/dev/apps/allocations/app/store/allocation.js). I'm more than happy to make this change if you'd prefer.

Reply:
Yeah, I'm not sure what the change is you're looking for. It would be great if you could make the update and I'll try to replicate your solution if the issue comes up again in the future.

Reply:
As in the link to the allocations store code in my comment, we want to utilize the rxjs libraries more effectively, since it provides some pretty awesome tooling for reshaping the data as needed, and converting to promises using the `toPromise` method instead of resolving inside a `subscribe` call. 

This requires gaining some familiarity with the [rxjs operators](https://rxjs.dev/api/operators) more generally, but it looks like all you'll need to reshape the data here is to add a [map](https://rxjs.dev/api/operators/map) operator inside the pipe method to reshape the data as you have done inside the subcribe call, and then swap out the `subscribe` for a `toPromise()`

Reply:
Is this more to your liking?

```const decorateVote = async (vote) => {
  const executionTargets = getVoteExecutionTargets(vote)
  if (!executionTargets.length) {
    // If there's no execution target, consider it targetting this Dot Voting app
    const currentApp = await app.currentApp().pipe(first()).toPromise()
    return {
      ...vote,
      executionTargetData: {
        ...currentApp,
        identifier: undefined,
      }
    }
  }```